mix.exs

```diff
@@ -0,0 +1,51 @@
+defmodule Gitea.MixProject do
+  use Mix.Project
+
+  def project do
+    [
+      app: :gitea,
+      version: "0.1.0",
+      elixir: "~> 1.12",
+      start_permanent: Mix.env() == :prod,
+      test_coverage: [tool: ExCoveralls],
+      preferred_cli_env: [
+        c: :test,
+        coveralls: :test,
+        "coveralls.html": :test,
+        "coveralls.json": :test
+      ],
+      deps: deps(),
+      description: "Elixir interface for Gitea server",
+      package: package()
+    ]
+  end
+
+  # Run "mix help compile.app" to learn about applications.
+  def application do
+    [
+      extra_applications: [:logger]
+    ]
+  end
+
+  # Run "mix help deps" to learn about dependencies.
+  defp deps do
+    [
+      {:git_cli, "~> 0.3.0"},
+      {:req, "~> 0.2.2"},
```

Comment:
Reading the `req` docs https://hexdocs.pm/req/Req.html / https://github.com/wojtekmach/req
I don't see an obvious way to mock requests 
And the tests seem incomplete: https://github.com/wojtekmach/req/blob/main/test/req_test.exs (to me ...)

Totally on-board with replacing [`HTTPoison`](https://github.com/edgurgel/httpoison) if there's a performance bottleneck; but we really need to _benchmark_ it somehow. 

Also went down the rabbit hole and read the source of:
+ Finch: https://github.com/sneako/finch
+ Nimble Pool: https://github.com/dashbitco/nimble_pool  Mostly written by José Valim: https://github.com/dashbitco/nimble_pool/graphs/contributors
+ Mint: https://github.com/elixir-mint/mint

All looks good. and it's great that there appears to be innovation/improvement in the `Elixir` ecosystem. 
But I'm scratching my head trying to understand why we need to switch to this library _now_ (`before`) we publish the package ... 🤷‍♂️